internal/usde/usde_test.go

```diff
@@ -372,14 +385,31 @@ func Test_ResourceNeedsUpdating(t *testing.T) {
 				return makeISBServiceDefinition(*newISBServiceSpec)
 			}(),
 			usdeConfig: config.USDEConfig{
-				DefaultUpgradeStrategy:      config.PPNDStrategyID,
-				PipelineSpecExcludedPaths:   []string{"vertices.source.something"},
-				ISBServiceSpecExcludedPaths: []string{"jetstream.containerTemplate.resources.limits"},
+				DefaultUpgradeStrategy:       config.PPNDStrategyID,
+				PipelineSpecDataLossFields:   []config.SpecDataLossField{{Path: "spec.vertices.source.something"}},
+				ISBServiceSpecDataLossFields: []config.SpecDataLossField{{Path: "spec.jetstream.containerTemplate.resources.limits"}},
 			},
 			namespaceConfig:       &config.NamespaceConfig{UpgradeStrategy: "pause-and-drain"},
 			expectedNeedsUpdating: true,
 			expectedStrategy:      apiv1.UpgradeStrategyApply,
 		},
+		{
+			name:          "isb test - include subfields",
+			newDefinition: isbServiceDefn,
+			existingDefinition: func() kubernetes.GenericObject {
+				newISBServiceSpec := defaultISBServiceSpec.DeepCopy()
+				newISBServiceSpec.JetStream.ContainerTemplate.Resources.Limits = v1.ResourceList{v1.ResourceMemory: newMemLimit}
+				return makeISBServiceDefinition(*newISBServiceSpec)
+			}(),
+			usdeConfig: config.USDEConfig{
+				DefaultUpgradeStrategy:       config.PPNDStrategyID,
+				PipelineSpecDataLossFields:   []config.SpecDataLossField{{Path: "spec.vertices.source.something"}},
+				ISBServiceSpecDataLossFields: []config.SpecDataLossField{{Path: "spec.jetstream.containerTemplate.resources.limits", IncludeSubfields: true}},
+			},
+			namespaceConfig:       &config.NamespaceConfig{UpgradeStrategy: "pause-and-drain"},
+			expectedNeedsUpdating: true,
+			expectedStrategy:      apiv1.UpgradeStrategyPPND,
+		},
```

Comment:
Do you have a test in here which includes the field specified in the new definition and not specified in the original definition? (or the other way around?) For this test, it looks like it's specified in both the `newDefinition` and `existingDefinition` if I'm not mistaken? By removing it from one or the other, we could test Dillen's bug.